pandas/util/_validators.py

```diff
@@ -324,13 +324,13 @@ def validate_percentile(q: float | Iterable[float]) -> np.ndarray:
     q_arr = np.asarray(q)
     # Don't change this to an f-string. The string formatting
     # is too expensive for cases where we don't need it.
-    msg = "percentiles should all be in the interval [0, 1]. Try {} instead."
+    msg = "percentiles should all be in the interval [0, 1]"
     if q_arr.ndim == 0:
         if not 0 <= q_arr <= 1:
-            raise ValueError(msg.format(q_arr / 100.0))
```

Comment:
Can we add a test for the new error message?

Reply:
yeah !! 

> Can we add a test for the new error message?


What about something like this ?

```python
import re
import numpy as np
import pandas as pd
import pytest


@pytest.fixture
def valid_percentiles():
    return [
        [0.0, 0.25, 0.5, 0.75, 1.0],
        (0.0, 0.25, 0.5, 0.75, 1.0),
        np.array([0.0, 0.25, 0.5, 0.75, 1.0])
    ]


@pytest.fixture
def invalid_percentiles():
    return [
        [-0.01, 1.01],
        (-0.01, 1.01),
        np.array([-0.01, 1.01])
    ]


def test_validate_percentile_valid_input(valid_percentiles):
    for percentiles in valid_percentiles:
        result = pd.util._validators.validate_percentile(percentiles)
        assert np.array_equal(result, np.array(percentiles))


def test_validate_percentile_invalid_input(invalid_percentiles):
    expected_error_msg = "percentiles should all be in the interval [0, 1]"
    for p in invalid_percentiles:
        with pytest.raises(ValueError, match=re.escape(expected_error_msg)):
            pd.util._validators.validate_percentile(p)


def test_validate_percentile_negative_array_input():
    percentile_array = np.array([-0.5, 0.25, 1.5])
    expected_error_msg = "percentiles should all be in the interval [0, 1]"
    with pytest.raises(ValueError, match=re.escape(expected_error_msg)):
        pd.util._validators.validate_percentile(percentile_array)
```

Reply:
Something like ``test_validate_percentile_negative_array_input`` is sufficient, you don't need the others I think.
can you trigger the error somehow from calling a public function? To check that the appropriate error is shown to the user.

Reply:
> can you trigger the error somehow from calling a public function? To check that the appropriate error is shown to the user.

something like this ?
```python

import numpy as np
import pandas as pd

def my_function(percentiles):
    valid_percentiles = pd.util._validators.validate_percentile(percentiles)

percentiles = np.array([-0.5, 0.25, 1.5])
my_function(percentiles)

```

Reply:
just call Series.quantile with the invalid values

Reply:
something like this ?

```python
import pandas as pd
import numpy as np

percentiles = [-0.5, 0.25, 1.5]
s = pd.Series(np.arange(10))
try:
    s.quantile(percentiles)
except ValueError as e:
    print(str(e))
```

Reply:
You can check the contributor guide on how we test exceptions (or check other tests)

Reply:
I apologize for the delay in my response.

Hope something like this ?

```python
import pandas as pd
import numpy as np
import pytest

def test_series_quantile_invalid_percentile():
    s = pd.Series(np.random.randn(100))
    percentile_array = [-0.5, 0.25, 1.5]
    with pytest.raises(ValueError, match="percentiles should all be in the interval [0, 1]"):
        s.quantile(percentile_array)
```

Reply:
Yep this looks good

Reply:
Sorry, but I wonder how can I put the things together ?

Reply:
Please read our contributing guide, there is an explanation how to add tests